test/compiler/irpasses.jl

```diff
@@ -1092,3 +1092,13 @@ let src = code_typed1(foo_defined_last_iter, Tuple{Int})
         end
     end
 end
+
+# Issue #47180, incorrect phi counts in CmdRedirect
+function a47180(b; stdout )
+    c = setenv(b, b.env)
+    if true
+        c = pipeline(c, stdout)
+    end
+    c
+end
+@test isa(a47180(``; stdout), Base.AbstractCmd)
```

Comment:
For what is worth, by playing a bit more with creduce (more specifically, searching for `"TypeError: in new"` instead of the full message `"TypeError: in new, expected Base.AbstractCmd, got a value of type Nothing"`, to give it more freedom) I've got a marginally simpler reproducer:
```julia
julia> function a(b)
           cmd = setenv(``, b.env)
           if true
               cmd = (cmd, )
           end
           cmd
       end
a (generic function with 1 method)

julia> a(``)
ERROR: TypeError: in new, expected Cmd, got a value of type Nothing
Stacktrace:
 [1] a(b::Cmd)
   @ Main ./REPL[1]:4
 [2] top-level scope
   @ REPL[2]:1
```
```suggestion
# Issue #47180, incorrect phi counts in CmdRedirect
function a47180(b)
    cmd = setenv(``, b.env)
    if true
        cmd = (cmd, )
    end
    cmd
end
@test isa(a47180(``), Tuple{Cmd})
```